packages/updating-element/src/updating-element.ts

```diff
@@ -926,3 +1021,55 @@ export abstract class UpdatingElement extends HTMLElement {
 // Apply polyfills if available
 // eslint-disable-next-line @typescript-eslint/no-explicit-any
 (globalThis as any)['updatingElementPlatformSupport']?.({UpdatingElement});
+
+// Dev mode warnings...
+if (DEV_MODE) {
+  const requestUpdateThenable = {
+    then: (
+      onfulfilled?: (value: boolean) => void,
+      _onrejected?: () => void
+    ) => {
+      console.warn(
+        `\`requestUpdate\` no longer returns a Promise.` +
+          `Use \`updateComplete\` instead.`
+      );
+      if (onfulfilled !== undefined) {
+        onfulfilled(false);
+      }
+    },
+  };
+
+  const requestUpdate = UpdatingElement.prototype.requestUpdate;
+  UpdatingElement.prototype.requestUpdate = function (
```

Comment:
Rather than patching `requestUpdate`, can you put the `if (DEV_MODE) { return ...` in `requestUpdate`?

Reply:
Done.

Reply:
Nit: This change means we leave a a `return void 0` in the production minified code that wasn't strictly needed before right?